blaze/interactive.py

```diff
@@ -224,7 +233,19 @@ def expr_repr(expr, n=10):
 
     # Scalars
     if ndim(expr) == 0 and isscalar(expr.dshape):
-        return repr(compute(expr))
+        result = compute(expr)
+        if 'float' in str(expr.dshape):
```

Comment:
I'm actually tempted to stick with `'float' in str(expr.dshape)`.  Given that we're under the `isscalar` check this is actually decently strong.  It's also fairly clear.